### ISSUE TITLE
Xapian audit mode for squatter

### ISSUE THREAD

**Issue by brong** [cyrusimap/cyrus-imapd#2375]
This came up with a FastMail customer report where search results were incomplete.

Records in xapian have a *G* key which matches the GUID in the cyrus.index record.

How to check!

Example code in search_engines.c: "search_update_mailbox".  Create a similar function "search_audit_mailbox".  It would need to use XAPINDEXED mode, because otherwise we don't see what's actually in the index, but it also needs to check the convindexed tables (cyrus.indexed.db) to know which messages to check.

The basic summary is:
while ((msg = mailbox_iter_step(iter))) {
   if (!is_in_cyrusindexeddb(msg)) continue;
   if (is_in_xapian(msg)) continue;
   unindexed_msgs++;
}

Obviously you don't want any unindexed messages!

**Comment by rsto** [comment]
@ajaysusarla I've started working on this in a feature branch: https://github.com/rsto/cyrus-imapd/commit/0dacdb6408ba6e0db3cfd0e80af74f281bf133f3 (https://github.com/rsto/cassandane/tree/xapian_audit_mode), but I'm not particularly happy with it, yet.

I thought, I could reuse the `is_indexed` callback on the search receivers, that I wrote a while back, but that's not really useful: firstly, it doesn't tell us about gaps between indexed.db and Xapian, secondly, it has side-effects (which is really painful).

I guess I'll add an audit callback to the search engine callbacks tomorrow, but feel free to provide any input or code!
